src/python/WMCore/Storage/StageOutMgr.py

```diff
@@ -211,141 +235,137 @@ def __call__(self, fileToStage):
                 self.completedFiles[fileToStage['LFN']] = fileToStage
                 if lfn in self.failed:
                     del self.failed[lfn]
 
                 logging.info("===> Stage Out Successful: %s", fileToStage)
-                fileToStage = stageoutPolicyReport(fileToStage, None, None, 'FALLBACK', 0)
+                fileToStage = stageoutPolicyReport(fileToStage, None, None, 'OVERRIDE', 0)
                 return fileToStage
             except Exception as ex:
-                fileToStage = stageoutPolicyReport(fileToStage, fallback.get('phedex-node', None),
-                                                   fallback['command'], 'FALLBACK', 60310)
+                fileToStage = stageoutPolicyReport(fileToStage, self.overrideConf['phedex-node'],\
+                    self.overrideConf['command'], 'OVERRIDE', 60310)
                 lastException = ex
-                continue
 
         raise lastException
-
-    def fallbackStageOut(self, lfn, localPfn, fbParams, checksums):
+    
+    
+    def stageOut(self, lfn, localPfn, checksums, stageOut_rfc=None):
         """
-        _fallbackStageOut_
-
-        Given the lfn and parameters for a fallback stage out, invoke it
-
-        parameters should contain:
+        _stageOut_
 
+        Given the lfn and a pair of stage out and corresponding Rucio file catalog, stageOut_rfc, or override configuration invoke the stage out
+        If use override configuration self.overrideConf should contain:
         command - the stage out impl plugin name to be used
         option - the option values to be passed to that command (None is allowed)
         lfn-prefix - the LFN prefix to generate the PFN
         phedex-node - the Name of the PNN to which the file is being xferred
-
-        """
-        pfn = "%s%s" % (fbParams['lfn-prefix'], lfn)
-
-        try:
-            impl = retrieveStageOutImpl(fbParams['command'])
-        except Exception as ex:
-            msg = "Unable to retrieve impl for fallback stage out:\n"
-            msg += "Error retrieving StageOutImpl for command named: "
-            msg += "%s\n" % fbParams['command']
-            raise StageOutFailure(msg, Command=fbParams['command'],
-                                  LFN=lfn, ExceptionDetail=str(ex))
-
-        impl.numRetries = self.numberOfRetries
-        impl.retryPause = self.retryPauseTime
-
-        try:
-            impl(fbParams['command'], localPfn, pfn, fbParams.get("option", None), checksums)
-        except Exception as ex:
-            msg = "Failure for fallback stage out:\n"
-            msg += str(ex)
-            raise StageOutFailure(msg, Command=fbParams['command'],
-                                  LFN=lfn, InputPFN=localPfn, TargetPFN=pfn)
-
-        return pfn
-
-    def localStageOut(self, lfn, localPfn, checksums):
         """
-        _localStageOut_
+        if not self.override:
+            command = stageOut_rfc[0]['command']
```

Comment:
if `stageOut_rfc` is `None` here, this will through an Error

Reply:
I add an exception handling